src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -296,19 +297,21 @@ const useGroupInfo = (channelId = null) => {
         );
         await userMember.saveOrUpdateIfExists(localDb);
       });
+
+      refresh('channelList');
+      refreshWithId('chat', channelId);
+      refresh('channelInfo');
+
+      setTimeout(() => {
+        setIsLoadingAddMember(false);
+        navigation.navigate('SignedChatScreen');
+      }, 500);
     } catch (e) {
+      setIsLoadingAddMember(false);
       console.log('error on memberSchema');
       console.log(JSON.stringify(e));
       console.log(e);
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes here are also correct. The `setIsLoadingAddMember(false)` is now placed inside both the `setTimeout` function and the catch block to ensure that the loading state is reset regardless of whether the operation was successful or not. However, it would be better to have a more robust error handling mechanism than just logging the error to the console.
```diff
+ setTimeout(() => {
+   setIsLoadingAddMember(false);
+   navigation.navigate('SignedChatScreen');
+ }, 500);
+ } catch (e) {
+   setIsLoadingAddMember(false);
+   console.log('error on memberSchema');
+   console.log(JSON.stringify(e));
+   console.log(e);
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->